llvm/lib/ProfileData/Coverage/CoverageMapping.cpp

```diff
@@ -223,9 +223,130 @@ Expected<int64_t> CounterMappingContext::evaluate(const Counter &C) const {
   return LastPoppedValue;
 }
 
+mcdc::TVIdxBuilder::TVIdxBuilder(const SmallVectorImpl<ConditionIDs> &NextIDs,
+                                 int Offset)
+    : Indices(NextIDs.size()) {
+  // Construct Nodes and set up each InCount
+  auto N = NextIDs.size();
+  SmallVector<MCDCNode> Nodes(N);
+  for (unsigned ID = 0; ID < N; ++ID) {
+    for (unsigned C = 0; C < 2; ++C) {
+#ifndef NDEBUG
+      Indices[ID][C] = INT_MIN;
+#endif
+      auto NextID = NextIDs[ID][C];
+      Nodes[ID].NextIDs[C] = NextID;
+      if (NextID >= 0)
+        ++Nodes[NextID].InCount;
+    }
+  }
+
+  // Sort key ordered by <-Width, Ord>
```

Comment:
Why end nodes of the decision need to be sorted in descending order of width?

Reply:
https://discourse.llvm.org/t/rfc-coverage-new-algorithm-and-file-format-for-mc-dc/76798#other-minor-implementations-5

I think it may be available for integrating branch and mcdc.

See also;
https://discourse.llvm.org/t/rfc-region-branch-coverage-by-bitmap/79629#branch-coverage-and-mcdc-test-vectors-4

Reply:
So motivation of the sort is to prepare for truncating all conditions with `width=1`, is it?
If so, why not just remove them then with an O(n) method?

Reply:
Why did you think they could be removed? It's the prerequisite to assign identical IDs between clang and llvm-cov.

Reply:
I see, if we did not sort them, we can not reduce size of bitmaps effectively since there might be other conditions with `width>1` getting large index.

I'm trying to remove the sort because I find it breaks some nature if the topological relationship of conditions is abnormal (e.g conditions from pattern matching in rust) and generates troubled index maps. I would investigate it more. Thanks!

Reply:
Interesting. Could you introduce me the discussion, or paste the problematic cond tree anywhere?

Reply:
:relaxed: Sorry I found I made a mistake. I previously encountered a example with decision tree:
```mermaid
flowchart TD
C0 -- T --> C1
C1 -- F --> F1
C1 -- T --> C3
C0 -- F --> C2
C2 -- T --> C3
C2 -- F --> F2
C3 -- F --> C4
C3 -- T --> T3
C4 -- T --> T4
C4 -- F --> F4
```
Normal boolean expressions can not generate such decision tree, where either true next of a condition is a next (true or false) of its false next, or false next of a condition is a next of its true next. 

I tried to disable the sort and fixed some cases. So I guessed something was broken by the sort. But that's wrong. The real problem is the way to update cond bitmap, which is specified for previous cond bitmap but is inappropriate for current. This implementation still works for pattern matching, thanks to your excellent efforts!

Reply:
At a glance, `F1` and `F2` have `w=1`, and others have `w=2`. Correct?

Reply:
Yes. The index map is right. The key difference is at most one of `C0` and `C2` can be true, which not likes `(C0 && C1) || C2`. In previous I updated false bit of `C2` when `C0` is true since if `C0` is true, `C2` must be false. While at present we can not do this.